### ISSUE TITLE
Explicitly specified type arguments should not be faded when they differ in nullability from inferred type arguments

### ISSUE THREAD

**Issue by jnm2** [dotnet/roslyn#45484]
**Version Used**: 16.6.2

No diagnostic should be shown here. The name cannot be simplified without adding at least two `!` suppressions due to the choice of a generic type argument (`string?`) that would not make sense to choose here in the first place.

`_` denotes a faded span.

```cs
using System;
using System.Collections.Generic;
using System.Diagnostics.CodeAnalysis;

#nullable enable

class C
{
    bool M(IEnumerable<string> s, [NotNullWhen(true)] out string? result)
    {
        // ℹ IDE0001 "Name can be simplified. Type argument specification is redundant."
        // 💡 Simplify name 'TryFirst<string>' 
        //               ________
        return s.TryFirst<string>(x => x.Length % 2 == 0, out result);
    }
}

static class Extensions
{
    public static bool TryFirst<T>(this IEnumerable<T> source, Func<T, bool> predicate, [MaybeNullWhen(false)] out T value)
    {
        throw new NotImplementedException();
    }
}
```

**Comment by mavasani** [comment]
@CyrusNajmabadi I wonder if this is due to the symbol equivalence comparer bug that you have a PR out for?

**Comment by CyrusNajmabadi** [comment]
i think it's unrelated.  it's effectively about this specific feature (simplification) having no knowledge about nullability.

**Comment by CyrusNajmabadi** [comment]
It's worth noting thatfrom teh *existing* original design of this system, things are working properly.  i.e. those type args are redundant *wrt* to runtime meaning.  Unfortunately, at hte time it was written, there was no concept that you could have warnings provided by how you explicitly provided/inferred types.  We'll have to update it to understand this.

**Comment by jasonmalinowski** [comment]
Odd since we've fixed bugs like this before: https://github.com/dotnet/roslyn/issues/37980
